File: src/types/model/transfer.ts
export interface Transfer {
  id: number;
  transfer_from: string;
  transfer_to: string;
  transfer_amount: number;
  transfer_time: string;
  created_at: string;
  updated_at: string;
}

export interface TransferTrashed {
  id: number;
  transfer_from: string;
  transfer_to: string;
  transfer_amount: number;
  transfer_time: string;
  created_at: string;
  updated_at: string;
  deleted_at: string;
}

export interface TransferMonthStatusSuccess {
  year: string;
  month: string;
  total_amount: number;
  total_success: number;
}

export interface TransferYearStatusSuccess {
  year: string;
  total_amount: number;
  total_success: number;
}

export interface TransferMonthStatusFailed {
  year: string;
  month: string;
  total_amount: number;
  total_failed: number;
}

export interface TransferYearStatusFailed {
  year: string;
  total_amount: number;
  total_failed: number;
}

export interface TransferMonthAmount {
  month: string;
  total_amount: number;
}

export interface TransferYearAmount {
  year: string;
  total_amount: number;
}

File: src-tauri/src/model/transfer.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize)]
pub struct TransferResponse {
    pub id: i32,
    pub transfer_no: String,
    pub transfer_from: String,
    pub transfer_to: String,
    pub transfer_amount: i32,
    pub transfer_time: String,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Serialize, Deserialize)]
pub struct TransferResponseDeleteAt {
    pub id: i32,
    pub transfer_no: String,
    pub transfer_from: String,
    pub transfer_to: String,
    pub transfer_amount: i32,
    pub transfer_time: String,
    pub created_at: String,
    pub updated_at: String,
    pub deleted_at: String,
}

#[derive(Serialize, Deserialize)]
pub struct TransferResponseMonthStatusSuccess {
    pub year: String,
    pub month: String,
    pub total_amount: i32,
    pub total_success: i32,
}

#[derive(Serialize, Deserialize)]
pub struct TransferResponseYearStatusSuccess {
    pub year: String,
    pub total_amount: i32,
    pub total_success: i32,
}

#[derive(Serialize, Deserialize)]
pub struct TransferResponseMonthStatusFailed {
    pub year: String,
    pub month: String,
    pub total_amount: i32,
    pub total_failed: i32,
}

#[derive(Serialize, Deserialize)]
pub struct TransferResponseYearStatusFailed {
    pub year: String,
    pub total_amount: i32,
    pub total_failed: i32,
}

#[derive(Serialize, Deserialize)]
pub struct TransferMonthAmountResponse {
    pub month: String,
    pub total_amount: i32,
}

#[derive(Serialize, Deserialize)]
pub struct TransferYearAmountResponse {
    pub year: String,
    pub total_amount: i32,
}
